### PR TITLE
Removing sleep from test steps

### PR DESCRIPTION
**Pull request by VeereshAradhya**
- Removed sleep step from auto prune and addon tests
- Improved verification part of rbac tests

### CODE REVIEW

File: pkg/oc/oc.go [hunk 1]
```diff
@@ -67,6 +73,45 @@ func UpdateTektonConfigwithInvalidData(patch_data, errorMessage string) {
 	}
 }
 
+func AssertCronjobPresent(c *clients.Clients, cronJobName, namespace string) {
```
Comment (ppitonak) [inline pkg/oc/oc.go:76]: Does it make sense to also check that there is only one cron job with given name?

File: pkg/operator/rbac.go [hunk 2]
```diff
@@ -82,12 +83,12 @@ func AssertClusterRole(clients *clients.Clients, clusterRoleName string) {
...
 
 func AssertServiceAccountAfterDisable(clients *clients.Clients, ns, targetSA string) {
-	log.Printf("Verifying that service account %s doesn't exists\n", targetSA)
 	err := wait.Poll(config.APIRetry, config.APITimeout, func() (bool, error) {
+		log.Printf("Verifying that service account %s doesn't exists\n", targetSA)
 		saList, err := clients.KubeClient.Kube.CoreV1().ServiceAccounts(ns).List(context.TODO(), metav1.ListOptions{})
 		for _, item := range saList.Items {
 			if item.Name == targetSA {
-				return false, fmt.Errorf("found serviceaccount %s/%s", ns, targetSA)
+				return false, err
```
Comment (ppitonak) [inline pkg/operator/rbac.go:91]: I don't understand this change, can you explain please? Can the `saList` and `err` contain data at the same time?

File: pkg/operator/rbac.go [hunk 3]
```diff
@@ -13,8 +13,9 @@ import (
 )
 
 func AssertServiceAccount(clients *clients.Clients, ns, targetSA string) {
```
Comment (ppitonak) [inline pkg/operator/rbac.go:15]: IMHO we should unify the naming conventions for these `Assert...` functions... I like the one that you've just added (`AssertCronjobPresent` and `AssertCronjobNotPresent`). I'm OK doing it in a new PR.

File: pkg/oc/oc.go (hunk 1 above)
Comment (VeereshAradhya) [reply pkg/oc/oc.go:76]: I thought about it, but the problem is if we have annotated any namespace with the pruner annotations and if there is any failure in deleting the namespace, there will be more than one cronjobs

File: pkg/operator/rbac.go (hunk 2 above)
Comment (VeereshAradhya) [reply pkg/operator/rbac.go:91]: I will explain this in a call

File: pkg/k8s/k8s.go [hunk 4]
```diff
@@ -354,3 +354,43 @@ func AssertPrunerCronjobWithContainer(c *clients.Clients, namespace, num string)
...
+		}
+		for _, cj := range cjs.Items {
+			if strings.Contains(cj.Name, cronJobName) {
+				return true, nil
+			}
+		}
+		return false, nil
+	})
+	if err != nil {
+		assert.FailOnError(fmt.Errorf("Expected: cronjob with prefix %v to be present in namespace %v, Actual:cronjob with prefix %v to be not present in namespace %v", cronJobName, namespace, cronJobName, namespace))
```
Comment (ppitonak) [inline pkg/k8s/k8s.go:373]: in both functions
* space after `Actual:`
* in different functions you are using `%v present` and `%v not present` which I like more

File: pkg/operator/rbac.go [hunk 5]
```diff
@@ -12,41 +12,47 @@ import (
 	"k8s.io/apimachinery/pkg/util/wait"
 )
 
-func AssertServiceAccount(clients *clients.Clients, ns, targetSA string) {
+func AssertServiceAccountPesent(clients *clients.Clients, ns, targetSA string) {
 
 	err := wait.Poll(config.APIRetry, config.APITimeout, func() (bool, error) {
-		log.Printf("Verifying that service account %s exists\n", targetSA)
+		log.Printf("Verifying that service account %s exist\n", targetSA)
```
Comment (ppitonak) [inline pkg/operator/rbac.go:18]: s/exist/exists/ in multiple places

**Pull request merged**